### ISSUE TITLE
raised md-buttons with primary colors not accessible for keyboard only + visually impaired users

### ISSUE THREAD

**Issue by opadillaintegrify** [angular/material#4627]
There are enough examples on https://material.angularjs.org/latest/#/demo/material.components.button
Use the keyboard to tab through the buttons on the page and look at it from the perspective of a visually impaired user who can't use a mouse...

Keyboard accessible web sites typically rely on the CSS outline on focus to let the user know which button currently has focus as the user tabs around.  Angular Material's CSS removed the outline of the buttons ( outline: 0; ), which some claim is a big no no from an accessibility standpoint unless you replace it with some other visual way of letting the user know when something has focus.

Angular Material does this by changing the hue somewhat on focus which is good from a keyboard nav standpoint, but the color change is far too subtle for the visually impaired.  Case in point, checkout the raised primary and warning buttons on that demo page...
```
<md-button class="md-raised md-primary">Primary</md-button>
<md-button class="md-raised md-warn">Warn</md-button>
```

**Comment by jelbourn** [comment]
I agree that the focus hue is far too subtle. Will consult with Material Design UX team on this.

**Comment by marcysutton** [comment]
Also see https://github.com/angular/material/issues/807.
